src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java

```diff
@@ -191,35 +208,47 @@ public TypeSpec createTypeAdapterFactory(ClassName className, ClassName autoValu
         .build();
   }
 
-  public TypeSpec createTypeAdapter(String className, String autoValueClassName, List<Property> properties) {
-    ClassName annotatedClass = ClassName.bestGuess(autoValueClassName);
+  public TypeSpec createTypeAdapter(ClassName className, ClassName autoValueClassName, List<Property> properties) {
     ClassName typeAdapterClass = ClassName.get(TypeAdapter.class);
-    ParameterizedTypeName superClass = ParameterizedTypeName.get(typeAdapterClass, annotatedClass);
+    ParameterizedTypeName superClass = ParameterizedTypeName.get(typeAdapterClass, autoValueClassName);
+
 
-    FieldSpec gsonField = FieldSpec.builder(Gson.class, "gson").build();
+    ImmutableMap<Property, FieldSpec> adapters = createFields(properties);
+
+    MethodSpec.Builder constructor = MethodSpec.constructorBuilder()
+            .addModifiers(PUBLIC)
+            .addParameter(Gson.class, "gson");
```

Comment:
Can you make this a `PropertySpec` to be referenced by `$N` in lines 226 and 230?